packages/@react-spectrum/s2/src/Toast.tsx

```diff
@@ -0,0 +1,579 @@
+   * @default "bottom"
+   */
+  placement?: ToastPlacement
+}
+
+export interface ToastOptions extends Omit<RACToastOptions, 'priority'>, DOMProps {
+  /** A label for the action button within the toast. */
+  actionLabel?: string,
+  /** Handler that is called when the action button is pressed. */
+  onAction?: () => void,
+  /** Whether the toast should automatically close when an action is performed. */
+  shouldCloseOnAction?: boolean
+}
+
+export interface SpectrumToastValue extends DOMProps {
+  /** The content of the toast. */
+  children: string,
+  /** The variant (i.e. color) of the toast. */
+  variant: 'positive' | 'negative' | 'info' | 'neutral',
+  /** A label for the action button within the toast. */
+  actionLabel?: string,
+  /** Handler that is called when the action button is pressed. */
+  onAction?: () => void,
+  /** Whether the toast should automatically close when an action is performed. */
+  shouldCloseOnAction?: boolean
+}
+
+function startViewTransition(fn: () => void, type: string) {
+  if ('startViewTransition' in document) {
+    // Safari doesn't support :active-view-transition-type() yet, so we fall back to a class on the html element.
+    document.documentElement.classList.add(toastCss[type]);
+    let viewTransition = document.startViewTransition({
+      update: () => flushSync(fn),
+      types: [toastCss[type]]
+    });
+
+    viewTransition.ready.catch(() => {});
+    viewTransition.finished.then(() => {
+      document.documentElement.classList.remove(toastCss[type]);
+    });
+  } else {
+    fn();
+  }
+}
+
+// There is a single global toast queue instance for the whole app, initialized lazily.
+let globalToastQueue: ToastQueue<SpectrumToastValue> | null = null;
+function getGlobalToastQueue() {
+  if (!globalToastQueue) {
+    globalToastQueue = new ToastQueue({
+      maxVisibleToasts: Infinity,
+      wrapUpdate(fn, action) {
+        startViewTransition(fn, `toast-${action}`);
+      }
+    });
+  }
+
+  return globalToastQueue;
+}
+
+function addToast(children: string, variant: SpectrumToastValue['variant'], options: ToastOptions = {}) {
+  let value = {
+    children,
+    variant,
+    actionLabel: options.actionLabel,
+    onAction: options.onAction,
+    shouldCloseOnAction: options.shouldCloseOnAction,
+    ...filterDOMProps(options)
+  };
+
+  // Minimum time of 5s from https://spectrum.adobe.com/page/toast/#Auto-dismissible
+  // Actionable toasts cannot be auto dismissed. That would fail WCAG SC 2.2.1.
+  // It is debatable whether non-actionable toasts would also fail.
+  let timeout = options.timeout && !options.actionLabel ? Math.max(options.timeout, 5000) : undefined;
+  let queue = getGlobalToastQueue();
+  let key = queue.add(value, {timeout, onClose: options.onClose});
+  return () => queue.close(key);
+}
+
+type CloseFunction = () => void;
+
+const SpectrumToastQueue = {
+  /** Queues a neutral toast. */
+  neutral(children: string, options: ToastOptions = {}): CloseFunction {
```

Comment:
We should consider this for S2: [Add support for string formatting in React Spectrum v3 toast](https://github.com/adobe/react-spectrum/issues/6038#top)

Reply:
perhaps, but maybe post this release